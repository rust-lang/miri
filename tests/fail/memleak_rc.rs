//@error-pattern: the evaluated program leaked memory
//@stderr-per-bitwidth
//@normalize-stderr-test: ".*â”‚.*" -> "$$stripped$$"

use std::cell::RefCell;
use std::rc::Rc;

struct Dummy(Rc<RefCell<Option<Dummy>>>);

fn main() {
    let x = Dummy(Rc::new(RefCell::new(None)));
    let y = Dummy(x.0.clone());
    *x.0.borrow_mut() = Some(y);
}
