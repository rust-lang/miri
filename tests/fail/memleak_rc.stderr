The following memory was leaked: ALLOC (Rust heap, size: 32, align: 8) {
    0x00 │ 01 00 00 00 00 00 00 00 01 00 00 00 00 00 00 00 │ ................
    0x10 │ 00 00 00 00 00 00 00 00 ╾$HEX[a1765]<TAG>─╼ │ ........╾──────╼
}

error: the evaluated program leaked memory

note: pass `-Zmiri-ignore-leaks` to disable this check

error: aborting due to previous error

